### PR TITLE
Pass options to task

### PR DESCRIPTION
**Pull request by Artur-**
## Description

Please list all relevant dependencies in this section and provide summary of the change, motivation and context.

Fixes # (issue)

## Type of change

- [ ] Bugfix
- [ ] Feature

## Checklist

- [ ] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [ ] I have added a description following the guideline.
- [ ] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [ ] New and existing tests are passing locally with my change.
- [ ] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java [hunk 1]
```diff
@@ -118,33 +118,22 @@ public NodeTasks(Options options) {
...
             if (packageUpdater != null && options.runNpmInstall) {
-                commands.add(new TaskRunNpmInstall(packageUpdater,
-                        options.enablePnpm, options.requireHomeNodeExec,
-                        options.nodeVersion, options.nodeDownloadRoot,
-                        options.useGlobalPnpm, options.nodeAutoUpdate,
-                        options.postinstallPackages));
+                commands.add(new TaskRunNpmInstall(packageUpdater, options));
 
                 commands.add(new TaskInstallWebpackPlugins(
                         new File(options.npmFolder, options.buildDirectory)));
```
Comment (caalador) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java:139]: This one is quite simple, but going away so I guess it is fine to keep it as is.

**Review by caalador — COMMENTED** [review]
Tasks inside methods are not updated.
Will they be updated in a next PR?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java (hunk 1 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java:139]: Yes I did not bother with that one but if we backport this to 23.3, maybe I should..

**Comment by Artur-** [comment]
Fixed remaining methods

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/Options.java [hunk 2]
```diff
@@ -616,18 +651,31 @@ public File getFrontendDirectory() {
...
+            return featureFlags;
+        }
+
+        final FeatureFlags flags = new FeatureFlags(lookup);
         if (javaResourceFolder != null) {
-            featureFlags.setPropertiesLocation(javaResourceFolder);
+            flags.setPropertiesLocation(javaResourceFolder);
         }
-        return featureFlags;
+        return flags;
```
[the remark this answers is not in the document]
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/Options.java]: Sure

**Review by caalador — DISMISSED** [review]

**Comment by caalador** [comment]
In the slow build you can see:
```
java.lang.NullPointerException: Cannot invoke "java.io.File.getAbsolutePath()" because the return value of "com.vaadin.flow.server.frontend.Options.getNpmFolder()" is null
  at com.vaadin.flow.server.frontend.TaskRunNpmInstall.runNpmInstall(TaskRunNpmInstall.java:251)
  at com.vaadin.flow.server.frontend.TaskRunNpmInstall.execute(TaskRunNpmInstall.java:145)
  at com.vaadin.flow.server.frontend.TaskRunNpmInstallTest.runNpmInstall_emptyDir_npmInstallIsExecuted(TaskRunNpmInstallTest.java:127)
```

**Comment by Artur-** [comment]
Maybe it works again now

**Review by caalador — DISMISSED** [review]

**Review by mcollovati — APPROVED** [review]

**Pull request merged**